### ISSUE TITLE
Move to catkin build system

### ISSUE THREAD

**Issue by dtmoodie** [tum-vision/lsd_slam#76]
Please port to catkin build system to work with newer ROS installs.

**Comment by icoderaven** [comment]
Can you please confirm that pull request #70 works for you?

**Comment by colek42** [comment]
your fork works for me on 14.04, indego.  Thanks!

**Comment by chrisspen** [comment]
It would be great if these were merged into master.

**Comment by chrisspen** [comment]
@icoderaven, Your fork doesn't compile for me on 14.04, Indigo. I get the error:

```
In file included from /home/chris/git/lsd_slam_test/src/lsd_slam/lsd_slam_viewer/src/main_viewer.cpp:25:0:
/home/chris/git/lsd_slam_test/src/lsd_slam/lsd_slam_viewer/src/PointCloudViewer.h:30:41: fatal error: lsd_slam_viewer/keyframeMsg.h: No such file or directory
 #include "lsd_slam_viewer/keyframeMsg.h"
                                         ^
compilation terminated.
In file included from /home/chris/git/lsd_slam_test/src/lsd_slam/lsd_slam_viewer/src/PointCloudViewer.cpp:22:0:
/home/chris/git/lsd_slam_test/src/lsd_slam/lsd_slam_viewer/src/PointCloudViewer.h:30:41: fatal error: lsd_slam_viewer/keyframeMsg.h: No such file or directory
 #include "lsd_slam_viewer/keyframeMsg.h"
```

**Comment by icoderaven** [comment]
Hm, this is a common issue with the message headers not compiling before the executables. I have a version of the CMakelists.txt that forces the build of the messages first. A temporary way to get it to build is to use a smaller number of parallel threads while building the package. I'll see if I can push that commit sometime.

**Comment by chrisspen** [comment]
Yeah, googling found me [this similar question](http://answers.ros.org/question/64448/trouble-with-catkin-compiling-node-and-generating-messages-in-same-package/), which mentions this bug in catkin. I worked around it by running catkin twice like:

```
catkin_make -j1
catkin_make
```

which feels a little hacky.

Yes, please push your fixed CMakeLists.txt when you're able.

**Comment by oblivione** [comment]
cd lsd_slam/

git checkout catkin 
you run " ls"

response should :

LICENSE lsd_slam lsd_slam_core lsd_slam_viewer README.md

Then add 
"build_depend>cmake_modules</build_depend"

"run_depend>cmake_modules</run_depend"

in "package.xml" both in the folder "lsd_slam/lsd_slam_viewer" and in the folder "lsd_slam/lsd_slam_core".

In "CMakeFiles.txt" both in the folder "lsd_slam/lsd_slam_viewer" and in the folder "lsd_slam/lsd_slam_core"

add:
find_package(cmake_modules REQUIRED)

check the line below (X11 may be mising ) 
target_link_libraries(lsdslam ${FABMAP_LIB} ${G2O_LIBRARIES} ${catkin_LIBRARIES} csparse cxsparse X11)

Then u can build your workspace by 
cd path_to_catkin_ws
catkin_make

Then again if got some error of header file missing then 
"CMakeFiles.txt" in the folder "lsd_slam/lsd_slam_core" add:
add_dependencies(lsdslam lsd_slam_viewer_generate_messages_cpp)
add_dependencies(live_slam lsd_slam_viewer_generate_messages_cpp)
add_dependencies(dataset lsd_slam_viewer_generate_messages_cpp)

"CMakeFiles.txt" in the folder "lsd_slam/lsd_slam_viewer" add:
add_dependencies(viewer lsd_slam_viewer_generate_messages_cpp)

**Comment by romdos** [comment]
I have the next error:
~/ros_workspace/catkin/src/lsd_slam/lsd_slam_core/src/GlobalMapping/g2oTypeSim3Sophus.h:24:34: fatal error: g2o/core/base_vertex.h: No such file or directory
compilation terminated.

**Comment by apesIITM** [comment]
@oblivione @JakobEngel @icoderaven  following the same steps, it gave this error [Eigen]:
=-=-=-=-=-=-=-=
`By not providing "FindEigen.cmake" in CMAKE_MODULE_PATH this project has
asked CMake to find a package configuration file provided by "Eigen", but
CMake did not find one.

Could not find a package configuration file provided by "Eigen" with any
of the following names:

EigenConfig.cmake
eigen-config.cmake

Add the installation prefix of "Eigen" to CMAKE_PREFIX_PATH or set
"Eigen_DIR" to a directory containing one of the above files. If "Eigen"
provides a separate development package or SDK, be sure it has been
installed.`
=-=-=-=-=-=-=-=

**Comment by romdos** [comment]
You must set a path to Eigen library.

**Comment by Toumi0812** [comment]
Hi @oblivione,
How to find  "path_to_catkin_ws" ?

Thanks

**Comment by 20chix** [comment]
> I have the next error:
> ~/ros_workspace/catkin/src/lsd_slam/lsd_slam_core/src/GlobalMapping/g2oTypeSim3Sophus.h:24:34: fatal error: g2o/core/base_vertex.h: No such file or directory
> compilation terminated.

I'm getting similar error did you fixed it?
